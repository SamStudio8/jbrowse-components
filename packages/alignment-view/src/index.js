import ViewType from '@gmod/jbrowse-core/pluggableElementTypes/ViewType'
import Plugin from '@gmod/jbrowse-core/Plugin'
import {
  ReactComponent as AlignmentViewReactComponent,
  stateModelFactory as alignmentViewStateModelFactory,
} from './AlignmentView'

export default class extends Plugin {
  install(pluginManager) {
    pluginManager.addViewType(
      () =>
        new ViewType({
          name: 'AlignmentView',
          stateModel: alignmentViewStateModelFactory(pluginManager),
          ReactComponent: AlignmentViewReactComponent,
        }),
    )
  }
}
