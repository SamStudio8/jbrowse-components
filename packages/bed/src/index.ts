import AdapterType from '@gmod/jbrowse-core/pluggableElementTypes/AdapterType'
import Plugin from '@gmod/jbrowse-core/Plugin'
import PluginManager from '@gmod/jbrowse-core/PluginManager'
import {
  AdapterClass as BigBedAdapterClass,
  configSchema as bigBedAdapterConfigSchema,
} from './BigBedAdapter'

export default class extends Plugin {
  install(pluginManager: PluginManager) {
    pluginManager.addAdapterType(
      () =>
        new AdapterType({
          name: 'BigBedAdapter',
          configSchema: bigBedAdapterConfigSchema,
          AdapterClass: BigBedAdapterClass,
        }),
    )
  }
}
