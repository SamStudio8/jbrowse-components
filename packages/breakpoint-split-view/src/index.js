export default class BreakpointSplitViewPlugin {
  install(pluginManager) {
    pluginManager.addViewType(() =>
      pluginManager.jbrequire(require('./BreakpointSplitView')),
    )
  }

  // eslint-disable-next-line @typescript-eslint/no-empty-function
  configure() {}
}
