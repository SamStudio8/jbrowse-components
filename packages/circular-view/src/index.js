export default class CircularViewPlugin {
  install(pluginManager) {
    pluginManager.addViewType(() =>
      pluginManager.jbrequire(require('./CircularView')),
    )

    pluginManager.addTrackType(() =>
      pluginManager.jbrequire(require('./StructuralVariantChordTrack')),
    )

    pluginManager.addRendererType(() =>
      pluginManager.jbrequire(require('./StructuralVariantChordRenderer')),
    )
  }

  // eslint-disable-next-line @typescript-eslint/no-empty-function
  configure() {}
}

export { default as ChordTrack } from './ChordTrack'
