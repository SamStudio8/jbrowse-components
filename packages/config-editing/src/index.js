import { lazy } from 'react'
import Plugin from '@gmod/jbrowse-core/Plugin'
import DrawerWidgetType from '@gmod/jbrowse-core/pluggableElementTypes/DrawerWidgetType'
import {
  reactComponent as ConfigurationEditorReactComponent,
  stateModelFactory as ConfigurationEditorStateModelFactory,
  configSchema as ConfigurationEditorConfigSchema,
  HeadingComponent as ConfigurationEditorHeadingComponent,
} from './ConfigurationEditorDrawerWidget'

export default class MainMenuBar extends Plugin {
  install(pluginManager) {
    pluginManager.addDrawerWidgetType(() => {
      return new DrawerWidgetType({
        name: 'ConfigurationEditorDrawerWidget',
        HeadingComponent: ConfigurationEditorHeadingComponent,
        configSchema: ConfigurationEditorConfigSchema,
        stateModel: ConfigurationEditorStateModelFactory(pluginManager),
        LazyReactComponent: lazy(() => ConfigurationEditorReactComponent),
      })
    })
  }
}

export {
  default as JsonEditor,
} from './ConfigurationEditorDrawerWidget/components/JsonEditor'
export {
  FileLocationEditor,
} from './ConfigurationEditorDrawerWidget/components/SlotEditor'
