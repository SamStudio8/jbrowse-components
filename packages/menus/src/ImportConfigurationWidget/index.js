import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'
import { ElementId } from '@gmod/jbrowse-core/util/types/mst'
import { types } from 'mobx-state-tree'

export const configSchema = ConfigurationSchema('ImportConfigurationWidget', {})

export const stateModel = types.model('ImportConfigurationWidget', {
  id: ElementId,
  type: types.literal('ImportConfigurationWidget'),
})

export const ReactComponent = import('./components/ImportConfigurationWidget')
