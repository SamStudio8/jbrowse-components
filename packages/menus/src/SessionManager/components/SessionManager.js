import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Icon from '@material-ui/core/Icon'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import ListSubheader from '@material-ui/core/ListSubheader'
import Paper from '@material-ui/core/Paper'
import { observer } from 'mobx-react-lite'

const useStyles = makeStyles(theme => ({
  root: {
    margin: theme.spacing(1),
  },
}))

export default observer(({ session }) => {
  const classes = useStyles()

  return (
    <Paper className={classes.root}>
      <List subheader={<ListSubheader>Choose a session to open</ListSubheader>}>
        {session.savedSessions.map(sessionSnapshot => {
          const { views = [] } = sessionSnapshot
          const openTrackCount = views.map(view => (view.tracks || []).length)
          let viewDetails
          switch (views.length) {
            case 0: {
              viewDetails = '0 views'
              break
            }
            case 1: {
              viewDetails = `1 view, ${openTrackCount[0]} open track${
                openTrackCount[0] === 1 ? '' : 's'
              }`
              break
            }
            case 2: {
              viewDetails = `2 views; ${openTrackCount[0]} and ${
                openTrackCount[1]
              } open tracks`
              break
            }
            default: {
              viewDetails = `${views.length} views; ${openTrackCount
                .slice(0, views.length - 1)
                .join(', ')}, and ${
                openTrackCount[views.length - 1]
              } open tracks`
              break
            }
          }
          return (
            <ListItem
              button
              onClick={() => session.activateSession(sessionSnapshot.name)}
              key={sessionSnapshot.name}
            >
              <ListItemIcon>
                <Icon>view_list</Icon>
              </ListItemIcon>
              <ListItemText
                primary={sessionSnapshot.name}
                secondary={viewDetails}
              />
            </ListItem>
          )
        })}
      </List>
    </Paper>
  )
})
