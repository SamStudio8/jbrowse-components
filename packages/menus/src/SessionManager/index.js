import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'
import { ElementId } from '@gmod/jbrowse-core/mst-types'
import { types } from 'mobx-state-tree'

export const configSchema = ConfigurationSchema('SessionManager', {})

export const stateModel = types.model('SessionManager', {
  id: ElementId,
  type: types.literal('SessionManager'),
})

export const ReactComponent = import('./components/SessionManager')
