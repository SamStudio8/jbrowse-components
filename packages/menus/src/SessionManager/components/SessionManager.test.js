import { createTestSession } from '@gmod/jbrowse-web/src/rootModel'
import React from 'react'
import { render } from 'react-testing-library'
import SessionManager from './SessionManager'

describe('<SessionManager />', () => {
  it('renders', () => {
    const session = createTestSession()
    const { container } = render(<SessionManager session={session} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
