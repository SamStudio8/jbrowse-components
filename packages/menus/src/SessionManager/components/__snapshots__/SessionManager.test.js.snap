// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SessionManager /> renders 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation1 makeStyles-root-1 MuiPaper-rounded"
>
  <ul
    class="MuiList-root MuiList-padding MuiList-subheader"
  >
    <li
      class="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
    >
      Choose a session to open
    </li>
  </ul>
</div>
`;
